src/psyclone/tests/psyir/nodes/call_test.py

```diff
@@ -628,6 +629,539 @@ def test_call_get_callees_local(fortran_reader):
+
+            assert len(arg_idx_list) == 3
+            assert arg_idx_list[0] == 0
+            assert arg_idx_list[1] == 2
+            assert arg_idx_list[2] == 1
+
+            assert result is routine_foo_b
+            print(" - Passed subtest foo_b[2]")
+
+    if 1:
+        routine_foo_c: Routine = root_node.walk(Routine)[3]
+        assert routine_foo_c.name == "foo_c"
+
+        if 1:
+            print("Subtest foo_c[0]:")
+
+            call_foo_c: Call = routine_main.walk(Call)[5]
+            assert call_foo_c.routine.name == "foo"
+
+            arg_idx_list = []
+            result: Routine = call_foo_c.get_callee(
+                                ret_arg_match_list=arg_idx_list)
+
+            print(f" - Found matching argument list: {arg_idx_list}")
+
+            assert len(arg_idx_list) == 3
+            assert arg_idx_list[0] == 0
+            assert arg_idx_list[1] == 1
+            assert arg_idx_list[2] == 2
+
+            assert result is routine_foo_c
+            print(" - Passed subtest foo_c[0]")
+
+        if 1:
+            print("Subtest foo_c[1]:")
+
+            call_foo_c: Call = routine_main.walk(Call)[6]
+            assert call_foo_c.routine.name == "foo"
+
+            arg_idx_list = []
+            result: Routine = call_foo_c.get_callee(
+                                ret_arg_match_list=arg_idx_list)
+
+            print(f" - Found matching argument list: {arg_idx_list}")
+
+            assert len(arg_idx_list) == 2
+            assert arg_idx_list[0] == 1
+            assert arg_idx_list[1] == 0
+
+            assert result is routine_foo_c
+            print(" - Passed subtest foo_c[1]")
+
+        if 1:
+            print("Subtest foo_c[2]:")
+
+            call_foo_c: Call = routine_main.walk(Call)[7]
+            assert call_foo_c.routine.name == "foo"
+
+            arg_idx_list = []
+            result: Routine = call_foo_c.get_callee(
+                                ret_arg_match_list=arg_idx_list)
+
+            print(f" - Found matching argument list: {arg_idx_list}")
+
+            assert len(arg_idx_list) == 3
+            assert arg_idx_list[0] == 1
+            assert arg_idx_list[1] == 0
+            assert arg_idx_list[2] == 2
+
+            assert result is routine_foo_c
+            print(" - Passed subtest foo_c[2]")
+
+
+def test_call_get_callee_7_matching_arguments_not_found(fortran_reader):
+    """
+    Trigger error that matching arguments were not found
+    """
+    code = """
+module some_mod
+  implicit none
+contains
+
+  subroutine main()
+    integer :: e, f, g
+    ! Use name 'd' which doesn't exist
+    call foo(e, f, d=g)
+  end subroutine
+
+  ! Matching routine
+  subroutine foo(a, b, c)
+    integer :: a, b, c
+  end subroutine
+
+end module some_mod"""
+
+    psyir = fortran_reader.psyir_from_source(code)
+
+    routine_main: Routine = psyir.walk(Routine)[0]
+    assert routine_main.name == "main"
+
+    call_foo: Call = routine_main.walk(Call)[0]
+
+    try:
```

Comment:
Again, you want `with pytest.raises(...) as err` here...

Reply:
Thanks. I'll also try to use this for other tests.